tests/api/test_sources_endpoint.py

```diff
@@ -0,0 +1,141 @@
+        assert client.post(
+            self.endpoint,
+            json.dumps(get_test_data()),
+            content_type='application/vnd.api+json'
+        ).status_code == 401
+
+    def test_improper_scope_post(self, client, share_user):
+        assert client.post(
+            self.endpoint,
+            json.dumps(get_test_data()),
+            content_type='application/vnd.api+json',
+            HTTP_AUTHORIZATION=share_user.authorization(),
+        ).status_code == 403
+
+    def test_successful_post_no_home_page(self, client, source_add_user):
+        test_data = get_test_data()
+        resp = client.post(
+            self.endpoint,
+            json.dumps(test_data),
+            content_type='application/vnd.api+json',
+            HTTP_AUTHORIZATION=source_add_user.authorization(),
+        )
+        resp_attributes = resp.data['attributes']
+        resp_related_share_user = resp.data['relationships']['share_user']['data']
+        resp_related_source_config = resp.data['relationships']['source_config']['data']['attributes']
+
+        created_label = resp_attributes['long_title'].replace(' ', '_').lower()
+        created_user = ShareUser.objects.get(pk=resp_related_share_user['id'])
+
+        assert resp.status_code == 202
+        assert resp_attributes['long_title'] == test_data['data']['attributes']['long_title']
+        assert resp_attributes['name'] == created_label
+        assert resp_attributes['home_page'] is None
+        assert resp_related_share_user['attributes']['username'] == created_label
+        assert resp_related_share_user['attributes']['authorization_token'] == created_user.accesstoken_set.first().token
+        assert created_user.is_trusted is True
+        assert resp_related_source_config['label'] == created_label
+
+    def test_successful_post_home_page(self, client, source_add_user):
+        test_data = get_test_data(home_page='http://test.homepage.net')
+        resp = client.post(
+            self.endpoint,
+            json.dumps(test_data),
+            content_type='application/vnd.api+json',
+            HTTP_AUTHORIZATION=source_add_user.authorization(),
+        )
+        resp_attributes = resp.data['attributes']
+
+        assert resp.status_code == 202
+        assert resp_attributes['long_title'] == test_data['data']['attributes']['long_title']
+        assert resp_attributes['home_page'] == test_data['data']['attributes']['home_page']
+
+    def test_bad_image_url(self, client, source_add_user):
+        test_data = get_test_data(icon=IMPROPER_ICON_URL)
+        resp = client.post(
+            self.endpoint,
+            json.dumps(test_data),
+            content_type='application/vnd.api+json',
+            HTTP_AUTHORIZATION=source_add_user.authorization(),
+        )
+
+        assert resp.status_code == 400
+        assert resp.data[0]['detail'] == 'Could not download/process image. ["Invalid type. Expected one of (\'image/png\', \'image/jpeg\'). Received text/html"]'
```

Comment:
nitpick: title should be `Could not download/process image` and detail should be `Invalid type....`